homeassistant/components/homekit/type_lights.py

```diff
@@ -179,7 +181,16 @@ def async_update_state(self, new_state):
 
         # Handle Color
         if CHAR_SATURATION in self.chars and CHAR_HUE in self.chars:
-            hue, saturation = new_state.attributes.get(ATTR_HS_COLOR, (None, None))
+            if ATTR_HS_COLOR in new_state.attributes:
+                hue, saturation = new_state.attributes[ATTR_HS_COLOR]
+            elif ATTR_COLOR_TEMP in new_state.attributes:
+                hue, saturation = color_temperature_to_hs(
```

Comment:
I must admit that I don't understand which situation we are in here but if there is any way to avoid `color_temperature_to_hs()`, we should – it gives horrible results.

Reply:
The HomeKit spec says that Color Temperature characteristic and the Hue&Saturation characteristics are mutually exclusive. If the bulb supports both, HomeKit converts color temp to an HS pair.  The accessory is expected to do the same.  Since Apple doesn't expose their formula, we are stuck with 
`color_temperature_to_hs` unless there is something better we can use.

Reply:
Oh, that's pretty bad but I guess there's not much that can be done about it. As an example, the HS color (28,61) from the test is something like pastel orange on my LIFX bulb, definitely not anything like an incandescent light.

Reply:
Could we give a look to the homebridge code since their integration works perfectly

Reply:
If you have a better solution, please open a PR to adjust the conversion code.